src/protocol/requests/request-builder.ts

```diff
@@ -0,0 +1,52 @@
+import { IllegalArgumentError } from '../exceptions';
+import { type RequestResponseType, type Request } from './request';
+
+export interface RequestBuilder<T extends Request<any>> {
+  getApiKey(): number;
+
+  build(correlationId: number, minVersion: number, maxVersion: number): T;
+}
+
+export type RequestBuilderResponseType<T extends RequestBuilder<any>> = RequestResponseType<ReturnType<T['build']>>;
+
+export abstract class RequestBuilderTemplate<T extends Request<any>> implements RequestBuilder<T> {
```

Comment:
Since there is already a RequestBuilder type, can these two be merged?

Reply:
I guess not